module-2/backend/week_four.md

```diff
@@ -10,14 +10,44 @@ Note: When you're done, submit a PR with a reflection in the comments about how
 ### Questions
 
 * What is a cookie?
+  - A cookie is a small file that lives on the client end. It is generated in the response cycle. It helps create the impression of "state" by passing back relevant information in the request process. This can help the server remember who we are and what we were up to last time our devices talked to eachother.
+  
 * What’s the difference between a session and a cookie?
+    - A cookie is is stored in the server/locally.
+    - A session is stored in the browser.
+
 * What’s a flash and when do you want to use flashes?
+  - A flash is a way to let your user know what is happening. This allows them to know if they completed a task satisfactorally or something failed. It's a way to conditionally send messages to your user when "stuff" happens.
+  
 * Why do people say “HTTP is stateless”?
+  - Becasue of the request response cycle. You are not "live and tapped into the server". You are making constant requests and getting constant responses.
+  
 * What’s authentication? Explain.
+  -  Authentication is checking that somebody is who they say you are.
+  
 * What’s the difference between authentication and authorization?
+  - Once again, Authentication is checking that somebody is who they say you are.
+  - Authorization make sure you are where you are supposed to be and not in places you don't belong.
+
 * What’s a before filter?
+  - It checks for authorization level allowing a user to do something. Still mildly fuzzy on this.
```

Comment:
It's a Rails tool that we use to run methods before controller actions. We've used it to check a user's authorization before allowing them to access different methods (actions) on our controllers, but it can be used to run *any* method before we run methods on our controllers. We used this again when we were setting our backpack before each method on our controller in the cart/order example.